### PR TITLE
Add instructions for rebuilding the image with a custom component branch

### PR DESCRIPTION
**Pull request by PromoFaux**
### **What does this PR aim to accomplish?:**

Title. Not much more to be said.

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -207,6 +207,17 @@ Why is this style of upgrading good?  A couple reasons: Everyone is starting fro
 
 To reconfigure Pi-hole you'll either need to use an existing container environment variables or if there is no a variable for what you need, use the web UI or CLI commands.
 
+### Building an image with alternative component branches
+
+Occasionally you may need to try an alternative branch of one of the components (`core`,`web`,`ftl`). On bare metal you would run, for example, `pihole checkout core custombranchname`, however in Docker world we have disabled this command as it can cause unpredictable results.
```
Comment (yubiuser) [inline README.md:212]: Where did we disable the checkout command?

Comment (rdwebdesign) [reply README.md:212]: This was actually reverted here: https://github.com/pi-hole/docker-pi-hole/pull/1101

Comment (yubiuser) [reply README.md:212]: Then Promo should decide if the want re-revert the change or change the Readme.

Comment (PromoFaux) [reply README.md:212]: #1101 was a v5 image change - I guess the `unsupportedFunc` stuff disappeared somewhere else for the v6 image as it was pretty much built from the ground up. I will add it in this PR

Comment (PromoFaux) [reply README.md:212]: Ah, it's because I got rid of `install.sh` in it's entirety.

Comment (yubiuser) [reply README.md:212]: Maybe the function is better placed in the `core` repo. Just check for the presence of the docker tag before executing the functions?

Comment (PromoFaux) [reply README.md:212]: We could add a check at the beginning of each function in the `core` repo to exit early if the docker tag is detected. That would achieve the same thing I think, at least here it is minimal patching and makes contextual sense?

Comment (PromoFaux) [reply README.md:212]: We should not rely on my memory. I had entirely forgotten about this:

https://github.com/pi-hole/pi-hole/pull/5018

So to answer @yubiuser's original question of "Where did we disable the checkout command?" ... In the above PR. I'll force away the second commit here

Comment (PromoFaux) [reply README.md:212]: And also I should read better. Checkout still needs to be disabled the same way. IGNORE EVERYTHING I AM WRITING. I'M HUNGOVER.

Comment (yubiuser) [reply README.md:212]: https://github.com/pi-hole/pi-hole/pull/5416

**Review by yubiuser — APPROVED** [review]

**Pull request merged**